src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -506,6 +509,10 @@ extern "C" void PalAttachThread(void* thread)
 extern "C" bool PalDetachThread(void* thread)
 {
     UNREFERENCED_PARAMETER(thread);
+    if (g_threadDetachCallback != nullptr)
```

Comment:
@jkotas My earlier plan was to reuse UnixEvent from PalRedhawkUnix.cpp for the "thread stopped" event.  Thread.Join would go to native code to wait for that event, bypassing WaitSubsystem.  The event would be created at pthread_create time and stored in the Thread instance as a SafeHandle, so it would have the same lifetime as the OS thread handle on Windows.  What do you think?